### DISCUSSION TITLE
Combining respectCache and treatAsUpdate for action run

### DISCUSSION THREAD

**Discussion by oyvindneu** [lostpebble/pullstate#139] [Q&A]
I want to update a list of notifications maximum every 15 seconds. 

So I set an intervall to use run with `respectCache = true` - this worked fine, but caused flickering of the result list, since finished was set to false while polling the notification api.

To solve this I added `treatAsUpdate: true, respectCache = true` but then useBeckon will never return updating = true.

The code works as I would expect if `respectCache: false, treatAsUpdate: true`. 

It seems like respectCache cannot be combined with treatAsUpdate - is there some other setting I have missed?

How is the intended way of solving this?

I can fix the code by only updating every 15 sec, but this would update the cache even if another event already had updated the cache (for instance if the user pressed a refresh button in the UI - onRefreshClick).

```typescript
  const [finished, result, updating] = getNotificationAction.useBeckon();
  console.log({ finished, result, updating });

  useEffect(() => {
    const interval = setInterval(() => {
      console.log('Update');
      getNotificationAction.run(undefined, { respectCache: true, treatAsUpdate: true });
    }, 1000);

    return () => clearInterval(interval);
  });
```

And the getNotificationAction have a cacheBreakHook to not update more frequently than.
```typescript
  {
    cacheBreakHook: ({ result, timeCached }) => !result.error && timeCached + 15 * 1000 < Date.now(),
  },
```

```
 const onRefreshClick = () => {
getNotificationAction.run(undefined, { treatAsUpdate: true });
}
```

**Comment by lostpebble** [comment]
Hi @oyvindneu ,

Have you tried out the option:

```ts
const [finished, result, updating] = TestAsyncAction.useBeckon(undefined, { holdPrevious: true });
```

It might solve your issue here- since it forces the action to hold the previous value until the cache is updated with the new one.
